trino/trino.go

```diff
@@ -976,7 +1046,16 @@ func (qr *driverRows) fetch(allowEOF bool) error {
 	qr.rowindex = 0
 	qr.data = qresp.Data
 	qr.nextURI = qresp.NextURI
+
+	qrStats := QueryProgressCallbackResponse{
+		ID:         qresp.ID,
+		QueryStats: qresp.Stats,
+	}
+
 	if len(qr.data) == 0 {
+		if qr.stmt.conn.callback != nil {
```

Comment:
You could move this check into the `updateCallback` function. You could also pass a reference to the whole response there and it should create the `QueryProgressCallbackResponse` only when necessary.